src/Parser/Parser.hs

```diff
@@ -74,11 +74,11 @@ addid n = do
   putState (PS c w (n:ids') x)
 
 -- | Get the names necessary to build 'While'
-getWhileNames :: Parser (Name, [Name])
+getWhileNames :: Parser [Name]
```

Comment:
~I think the get/putWhileNames functions should be renamed, and have their associated haddock comments updated. They're no longer just for updating the names used in a while expr, and these should reflect that.~ Scratch that after reading more into it.